wamr-compiler/CMakeLists.txt

```diff
@@ -132,21 +138,11 @@ endif ()
 
 message ("-- Build as target ${WAMR_BUILD_TARGET}")
 
-if (CMAKE_SIZEOF_VOID_P EQUAL 8)
-    if (WAMR_BUILD_TARGET STREQUAL "X86_64" OR WAMR_BUILD_TARGET STREQUAL "AMD_64"
-        OR WAMR_BUILD_TARGET MATCHES "AARCH64.*" OR WAMR_BUILD_TARGET MATCHES "RISCV64.*")
-    if (NOT WAMR_BUILD_PLATFORM STREQUAL "windows")
-      # Add -fPIC flag if build as 64-bit
-      set (CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -fPIC")
-      set (CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fPIC")
-      set (CMAKE_SHARED_LIBRARY_LINK_C_FLAGS "${CMAKE_SHARED_LIBRARY_LINK_C_FLAGS} -fPIC")
-      set (CMAKE_SHARED_LIBRARY_LINK_CXX_FLAGS "${CMAKE_SHARED_LIBRARY_LINK_CXX_FLAGS} -fPIC")
-    endif ()
-  else ()
-    add_definitions (-m32)
-    set (CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} -m32")
-    set (CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} -m32")
-  endif ()
+# Add -m32 flag if compiling on 64-bit system for 32-bit x86 target
```

Comment:
I noticed that the information regarding *Add -fPIC flag if build as 64-bit* has been deleted. Was this done on purpose?

Reply:
Yes, it is set by cmake now. I added POSITION_INDENPENDENT_CODE property to remaining libs: vmlib and aotclib on line 377

Reply:
It will be set if it is supported by compiler and it will use appropriate flag for that compiler. So it will be set in more cases than in the original code.